iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ShuntCompensatorAdderImpl.java

```diff
@@ -18,7 +18,9 @@
 
     private final VoltageLevelExt voltageLevel;
 
-    private ShuntCompensatorModelWrapper model;
+    private ShuntCompensatorNonLinearModelAdderImpl shuntCompensatorNonLinearModelAdder;
+
+    private ShuntCompensatorLinearModelAdderImpl shuntCompensatorLinearModelAdder;
```

Comment:
This is not sufficient to ensure someone don't set two different models. You should introduce a common interface or reset the corresponding attribute in the add methods.